### PR TITLE
Test on arm64 runners

### PR DESCRIPTION
**Pull request by farshidtz**

### CODE REVIEW

**Comment by locnnil** [comment]
@farshidtz, the issue with those arm64 runners is, in fact, related to [this bug on Launchpad](https://bugs.launchpad.net/ubuntu/+source/linux/+bug/1953613).

The kernel of the runner machine, [6.5.0-1015-azure](https://github.com/canonical/matter-pi-gpio-commander/actions/runs/8098014739/job/22130434349#step:3:99), doesn't have the option [GPIO_CDEV_V1](https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/tree/drivers/gpio/Kconfig#n87) enabled, as you can see here in [this action log](https://github.com/canonical/matter-pi-gpio-commander/actions/runs/8100280392/job/22137904319#step:5:600). Therefore, when using the `gpiod` utilities, we encounter an error indicating the incompatibility between API V1 and V2 of `libgpiod`.

This issue doesn't occur with the Raspberry Pi, for example:
```bash
pi4@alsvid:~$ cat /boot/config-`uname -r` | grep GPIO_CDEV
CONFIG_GPIO_CDEV=y
CONFIG_GPIO_CDEV_V1=y
pi4@alsvid:~$ 
```

However, the main question isn't about the kernel configuration but the Debian packages themselves, since this kernel does [accept the char dev interface](https://github.com/canonical/matter-pi-gpio-commander/actions/runs/8100610887/job/22139007093#step:5:599). Since API V1 is considered deprecated, the problem lies in the versions of those packages, as shown in [this action log](https://github.com/canonical/matter-pi-gpio-commander/actions/runs/8100610887/job/22139007093#step:5:783). The versions are 1.6.3.

We could resolve this by cross-compiling our own `libgpiod` with the latest version or creating a Debian package ourselves and installing it.

**Review by locnnil — COMMENTED** [review]
Another thing from this error: 

On the first issue observed on this [action output](https://github.com/canonical/matter-pi-gpio-commander/actions/runs/8065704750/job/22032084740#step:4:510), we got:
```bash
    snap_test.go:202: [exec] sudo matter-pi-gpio-commander.test-blink
    exec.go:140: [stdout] GPIO: 4
    exec.go:140: [stdout] GPIOCHIP: 0
    exec.go:140: [stderr] Failed to get gpio line: 4
```
Which leads to the problem being originated from the function [gpiod_chip_get_line()](https://github.com/canonical/matter-pi-gpio-commander/blob/a66a4967f6847807017690a32a21564a7493a4b2/test-blink.cpp#L63). If you take a look at the current libgpiod source code, you can see that this function [doesn't exist anymore](https://git.kernel.org/pub/scm/libs/libgpiod/libgpiod.git/tree/include/gpiod.h#n202).

In other words, the library installed on the snap recognizes this function, but the kernel is unable to respond its `syscall` due to the fact that it doesn't have the retroactive support for API V1 enabled.
Which means that our code, is based on API V1, and we should make some changes in order of this work with the API v2 of `libgpiod`.

**Comment by locnnil** [comment]
This PR is duplicated on #57, can we close it @farshidtz?
